memeolist/app/src/main/java/org/aerogear/android/app/memeolist/ui/MemeFormActivity.java

```diff
@@ -174,21 +164,23 @@ private void saveMeme(String imageUrl) {
                 .photourl(createMemeUrl(imageUrl))
                 .build();
 
-        apolloClient.mutate(mutation)
-                .enqueue(new ApolloCall.Callback<CreateMemeMutation.Data>() {
+        SyncClient
```

Comment:
[Idea] It feels like really bad code to have all data layer inside Activity. Do you think we could get that out in future?

Reply:
So for example apps I like as much code as rational being in the activity *IF* we are keeping the code to a minimum and it doesn't introduce bugs.

I think some of it can also be cleaned up with changes at the SDK level